components/layout/layout.service.ts

```diff
@@ -1,197 +1,191 @@
- * @param HsEventBusService
- * @param $window
- * @param $document
- * @param $timeout
- * @param $log
- */
+import {HsConfig} from '../../config.service';
+import {HsEventBusService} from '../core/event-bus.service';
+import {HsLogService} from '../../common/log/log.service';
+import {Injectable} from '@angular/core';
+@Injectable({
+  providedIn: 'root',
+})
 export class HsLayoutService {
-  constructor(HsConfig, HsEventBusService, $window, $document, $timeout, $log) {
-    'ngInject';
-    Object.assign(this, {
-      HsConfig,
-      HsEventBusService,
-      $window,
-      $document,
-      $timeout,
-      $log,
-    });
-
-    this.data = {
-      panels: [
-        {
-          enabled: true,
-          order: 0,
-          title: 'Map Compositions',
-          description: 'List available map compositions',
-          name: 'composition_browser',
-          directive: 'hs.compositions',
-          mdicon: 'map',
-        },
-        {
-          enabled: true,
-          order: 1,
-          title: 'Manage and Style Layers',
-          description: 'Manage and style your layers in composition',
-          name: 'layermanager',
-          directive: 'hs.layermanager',
-          mdicon: 'layers',
-        },
-        {
-          enabled: true,
-          order: 2,
-          title: 'Legend',
-          description: 'Display map legend',
-          name: 'legend',
-          directive: 'hs.legend',
-          mdicon: 'format_list_bulleted',
-        },
-        {
-          enabled: () => {
-            return this.panelEnabled('datasource_selector');
-          },
-          order: 3,
-          title:
-            angular.isUndefined(HsConfig.datasources) ||
-            HsConfig.datasources.length > 0
-              ? 'Datasource Selector'
-              : 'Add layers',
-          description: 'Select data or services for your map composition',
-          name: 'datasource_selector',
-          directive: 'hs.datasource-selector',
-          mdicon: 'dns',
-        },
-        {
-          enabled: true,
-          order: 5,
-          title: 'Measurements',
-          description: 'Measure distance or area at map',
-          name: 'measure',
-          directive: 'hs.measure',
-          mdicon: 'straighten',
-        },
-        {
-          enabled: true,
-          order: 6,
-          title: 'Print',
-          description: 'Print map',
-          name: 'print',
-          directive: 'hs.print',
-          mdicon: 'print',
-        },
-        {
-          enabled: true,
-          order: 7,
-          title: 'Share map',
-          description: 'Share map',
-          name: 'permalink',
-          directive: 'hs.permalink',
-          mdicon: 'share',
-        },
-        {
-          enabled: true,
-          order: 8,
-          title: 'Save composition',
-          description: 'Save content of map to composition',
-          name: 'saveMap',
-          directive: 'hs.save-map',
-          mdicon: 'save',
+  data = {
```

Comment:
This object seems to be deprecated

Reply:
True, it was in the material design template. Deleted